packages/sanity/src/_internal/cli/server/__tests__/buildVendorDependencies.test.ts

```diff
@@ -0,0 +1,255 @@
+import fs from 'node:fs'
+import path from 'node:path'
+
+import {beforeEach, describe, expect, it, jest} from '@jest/globals'
+
+import {buildVendorDependencies} from '../buildVendorDependencies'
+
+// Mocking the vite.build function to inspect its calls. This allows us to test
+// that the function is called with the correct configuration without actually
+// performing the build process.
+jest.mock('vite', () => ({build: jest.fn()}))
+
+describe('buildVendorDependencies', () => {
+  const {build} = require('vite')
+  const packageRoot = path.resolve(__dirname, '../../../../..')
+  const examplesRoot = path.join(packageRoot, './fixtures/examples')
+  const outputDir = './output-dir'
+
+  beforeEach(() => {
+    jest.resetAllMocks()
+    const vite = require('vite')
+    vite.build.mockResolvedValue({})
+  })
+
+  it('should throw if there is no matching entry in VENDOR_IMPORTS', () => {
+    const cwd = path.join(examplesRoot, 'prj-with-styled-components-5')
+
+    return expect(buildVendorDependencies({cwd, outputDir})).rejects.toThrow(
+      "Package 'styled-components' requires at least 6.1.0.",
+    )
+  })
+
+  it('should return the expected entry points for react 18', async () => {
+    const cwd = path.join(examplesRoot, 'prj-with-react-18')
+    const imports = await buildVendorDependencies({cwd, outputDir})
+
+    expect(imports).toEqual({
+      'react': '/vendor/react/index.mjs',
+      'react-dom': '/vendor/react-dom/index.mjs',
+      'react-dom/client': '/vendor/react-dom/client.mjs',
+      'react-dom/package.json': '/vendor/react-dom/package.json.mjs',
```

Comment:
❓ What's the reason to have `package.json` in the import map?

Reply:
it's technically allowed by the export map so that means it's technically possible for the user to import `react/package.json` in their code (or for a dependency to do it) so it would be a bug on us if we didn't support it.